lib/lambda_ethereum_consensus/state_transition/accessors.ex

```diff
@@ -205,45 +205,60 @@ defmodule LambdaEthereumConsensus.StateTransition.Accessors do
 
     case {get_block_root(state, data.target.epoch), get_block_root_at_slot(state, data.slot)} do
       {{:ok, block_root}, {:ok, block_root_at_slot}} ->
-        is_matching_source = data.source == justified_checkpoint
-        is_matching_target = is_matching_source && data.target.root == block_root
-        is_matching_head = is_matching_target && data.beacon_block_root == block_root_at_slot
-
-        if is_matching_source do
-          source_indices =
-            if is_matching_source &&
-                 inclusion_delay <= Math.integer_squareroot(ChainSpec.get("SLOTS_PER_EPOCH")) do
-              [Constants.timely_source_flag_index()]
-            else
-              []
-            end
-
-          target_indices =
-            if is_matching_target &&
-                 inclusion_delay <= ChainSpec.get("SLOTS_PER_EPOCH") do
-              [Constants.timely_target_flag_index()]
-            else
-              []
-            end
-
-          head_indices =
-            if is_matching_head &&
-                 inclusion_delay == ChainSpec.get("MIN_ATTESTATION_INCLUSION_DELAY") do
-              [Constants.timely_head_flag_index()]
-            else
-              []
-            end
-
-          {:ok, source_indices ++ target_indices ++ head_indices}
-        else
-          {:error, "Attestation source does not match justified checkpoint"}
-        end
+        determine_flag_indices(
```

Comment:
maybe, instead of this function that doesn't really add anything, you could have 3 helper functions

compute_source_indices
compute_target_indices
compute_head_indices

Reply:
Cool, just refactored it